### PR TITLE
When unblocking all contacts response should not contain jids

### PR DESCRIPTION
**Pull request by bartekgorny**
This PR addresses the Trello card: "Blocking Command: Bad push received when unblock all"

As stated in XEP-0191, point 3.5, when a user sends an "unblock" stanza without specifying jids to unblock, the server should unblock all contacts and reply with an empty result (no jid list). This PR fixes this.

### CODE REVIEW

**Comment by michalwski** [comment]
> This PR addresses the Trello card: "Blocking Command: Bad push received when unblock all"

Could you please create issue here from the card and reference to it? Or just put the examples from the card in this PR.

**Comment by michalwski** [comment]
Also dialyzer has some suggestions regarding types.

File: test/ejabberd_tests/tests/mod_blocking_SUITE.erl [hunk 1]
```diff
@@ -529,10 +534,10 @@ get_blocklist_items(Items) ->
 user_blocks(Blocker, Blockees) when is_list(Blockees) ->
     BlockeeJIDs = [ escalus_utils:jid_to_lower(escalus_client:short_jid(B)) || B <- Blockees ],
     AddStanza = block_users_stanza(BlockeeJIDs),
-    ct:pal("add stanza ~p", [AddStanza]),
+%%    ct:pal("add stanza ~p", [AddStanza]),
```
Comment (michalwski) [inline test/ejabberd_tests/tests/mod_blocking_SUITE.erl]: Let's remove these commented `ct:pal` calls.

**Comment by bartekgorny** [comment]
According to http://xmpp.org/extensions/xep-0191.html#unblockall

When I unblock all sending:

```
<iq id='6w6t1-62' type='set'><unblock xmlns='urn:xmpp:blocking'/></iq>
```

I should receive a push like this:

```
<iq from='gardano@erlang-solutions.com' to='gardano@erlang-solutions.com/9bb69f6dacb33248' id='push' type='set'><unblock xmlns='urn:xmpp:blocking'/></iq>
```

But I receive this push:

```
<iq from='gardano@erlang-solutions.com' to='gardano@erlang-solutions.com/9bb69f6dacb33248' id='push' type='set'><unblock xmlns='urn:xmpp:blocking'><item jid='griveroa-inaka@erlang-solutions.com'/><item jid='ramabit@erlang-solutions.com'/></unblock></iq>
```

Where `griveroa-inaka@erlang-solutions.com` and `ramabit@erlang-solutions.com` are the JIDs unblocked

**Pull request merged**